### PR TITLE
First CCD version

### PR DESCRIPTION
**Pull request by w-samuelgauthier**
## Description

Adds a first version of the CCD to the supported single devices.

## Related Issue

#16 

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [X] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [X] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/ThatsTheEnd/horiba-python-sdk/blob/master/CODE_OF_CONDUCT.md) document.
- [X] I've read the [`CONTRIBUTING.md`](https://github.com/ThatsTheEnd/horiba-python-sdk/blob/master/CONTRIBUTING.md) guide.
- [X] I've updated the code style using `make codestyle`.
- [X] I've written tests for all new methods and classes that I created.
- [X] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Comment by ThatsTheEnd** [comment]
Hi Samuel,
very nice work! I added synchronous examples because horiba requested them.
Should we add logging in the appropriate places?
On the hardware computer I have the problem that the ICL starts no matter whether I request it to start in the example or not.
I put a comment on the code.
Cheers,
Niko

File: examples/asynchronous_examples/ccd_example [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import asyncio
+
+from horiba_sdk.devices.device_manager import DeviceManager
+from horiba_sdk.devices.single_devices.ccd import ChargeCoupledDevice
+
+
+async def main():
+    device_manager = DeviceManager(start_icl=True)
```
Comment (ThatsTheEnd) [inline examples/asynchronous_examples/ccd_example:8]: No matter whether I set this to true or false, the icl starts on the computer that has the hardware connected.

Comment (w-samuelgauthier) [reply examples/asynchronous_examples/ccd_example]: This should be fixed now. The ICL will receive the shutdown command and killed if still running after this order. Maybe we need to think if this is the right way.

**Comment by w-samuelgauthier** [comment]
I added an error handling that loads the json file with all the ICL errors. The handler is created in the `DeviceManager`. Each time a response is received from the ICL, the errors are logged according their level (I just did the CCD and will do the Monochromator on another branch).
There is an internal mapping of error levels and log levels which is needed, because the level names may not be the same between the json file and the `loguru` library

File: horiba_sdk/devices/single_devices/ccd.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        print(await charge coupled device.is_open)
+        await charge coupled device.close()
+
+    """
+
+    def __init__(self, device_manager: DeviceManager) -> None:
+        super().__init__(device_manager)
+        self.device_list: list[str] = []
+
+    async def __aenter__(self) -> 'ChargeCoupledDevice':
```
Comment (w-samuelgauthier) [inline horiba_sdk/devices/single_devices/ccd.py:40]: We need to think if it's worth to leave those async context manager functions, as the `open` function changed.

Comment (w-samuelgauthier) [reply horiba_sdk/devices/single_devices/ccd.py:47]: Handled in separate task

**Review by ThatsTheEnd — APPROVED** [review]

**Pull request merged**